diff --git a/CMakeLists.txt b/CMakeLists.txt
index 42cd3049..2d8b0353 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -176,7 +176,11 @@ else()
   set(LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}" CACHE PATH "Directory where lib will install")
 
   set(QCA_PREFIX_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}" CACHE PATH "Directory where qca will install")
-  set(QCA_PLUGINS_INSTALL_DIR "${LIB_INSTALL_DIR}/${QCA_LIB_NAME}" CACHE PATH "Directory where qca plugins will install")
+  if(WIN32)
+    set(QCA_PLUGINS_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/bin/${QCA_LIB_NAME}" CACHE PATH "Directory where qca plugins will install")
+  else(WIN32)
+    set(QCA_PLUGINS_INSTALL_DIR "${LIB_INSTALL_DIR}/${QCA_LIB_NAME}" CACHE PATH "Directory where qca plugins will install")
+  endif()
   set(QCA_BINARY_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/bin" CACHE PATH "Directory where qca plugins will install")
   set(QCA_LIBRARY_INSTALL_DIR "${LIB_INSTALL_DIR}" CACHE PATH "Directory where qca library will install")
   set(QCA_FEATURE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/mkspecs/features" CACHE PATH "Directory where qca feature file will install")
@@ -209,6 +213,7 @@ foreach(PATH QCA_PLUGINS_INSTALL_DIR
              QCA_LIBRARY_INSTALL_DIR
              QCA_FEATURE_INSTALL_DIR
              QCA_INCLUDE_INSTALL_DIR
+             QCA_FULL_INCLUDE_INSTALL_DIR
              QCA_PRIVATE_INCLUDE_INSTALL_DIR
              QCA_DOC_INSTALL_DIR
              QCA_MAN_INSTALL_DIR
@@ -250,6 +255,7 @@ if(USE_RELATIVE_PATHS)
                QCA_LIBRARY_INSTALL_DIR
                QCA_FEATURE_INSTALL_DIR
                QCA_INCLUDE_INSTALL_DIR
+               QCA_FULL_INCLUDE_INSTALL_DIR
                QCA_PRIVATE_INCLUDE_INSTALL_DIR
                QCA_DOC_INSTALL_DIR
                QCA_MAN_INSTALL_DIR
